ios/brave-ios/Sources/Brave/Frontend/Browser/Toolbars/BottomToolbar/Menu/History/HistoryModel.swift

```diff
@@ -96,14 +97,46 @@ class HistoryModel: NSObject, ObservableObject {
       refreshHistory()
     }
 
-    listener = api?.add(self)
+    listener = api?.add(
+      HistoryStateObserver { [weak self] event in
+        guard let self = self else { return }
+
+        defer {
+          refreshHistory()
+        }
+
+        if case .modelLoaded = event {
+          if api?.isBackendLoaded == true {
+            listener?.destroy()
+            listener = nil
+
+            if isHistoryServiceLoaded {
+              return
+            }
+
+            isHistoryServiceLoaded = true
+            return
```

Comment:
Maybe i'm not seeing it, but it seems like this is equivalent to the following:

```
listener?.destroy()
listener = nil
isHistoryServiceLoaded = true
```

Which you already do below

Maybe the order matters or maybe setting `isHistoryServiceLoaded = true` when it's already set to true causes problems?